main_routers/memory_router.py

```
@@ -0,0 +1,253 @@
+                "response_metadata": {},
+                "type": t,
+                "name": None,
+                "id": None,
+                "example": False,
+                **({"tool_calls": [], "invalid_tool_calls": [], "usage_metadata": None} if t == "ai" else {})
+            }
+        })
+    try:
+        with open(file_path, 'w', encoding='utf-8') as f:
+            json.dump(arr, f, ensure_ascii=False, indent=2)
+        return {"success": True}
```

Ruff (0.14.8)

Consider moving this statement to an `else` block

(TRY300)